src/datasets/arrow_writer.py

```diff
@@ -671,8 +672,9 @@ def finalize(self, metrics_query_result: dict):
 
         from .utils import beam_utils
 
+        # Beam FileSystems require the system's path separator in the older versions
         shards_metadata = list(
-            beam.io.filesystems.FileSystems.match([self._parquet_path + "*.parquet"])[0].metadata_list
+            beam.io.filesystems.FileSystems.match([str(Path(self._parquet_path)) + "*.parquet"])[0].metadata_list
```

Comment:
`Path` replaces `://` with `:/` no ? This can be an issue for those writing datasets in a cloud storage

```python
>>> from pathlib import Path
>>> Path("gs://foo")
PosixPath('gs:/foo')
```

Maybe do the change only if it's a local path ?

Reply:
Fixed :)